### DISCUSSION TITLE
ort docker Scanner fails resolving provenance but ort built from sources doesn't

### DISCUSSION THREAD

**Discussion by qequ** [oss-review-toolkit/ort#6699] [Q&A] [answered]
I'm testing running ort on a repo that is hosted privately in gitlab and I'm having issues when running two separate versions of the scanner; one running from the sources with
```
./cli/build/install/ort/bin/ort scan -i ~/private-repo/analyzer-result.yml -o ~/private-repo/

```
and with docker built with [this instructions](https://github.com/oss-review-toolkit/ort/blob/main/docs/hints-for-use-with-docker.md)
```
 docker run  -v $PWD/:/project ort --info scan -i  /project/output/analyzer-result.yml -o /project/output

```

with the version of  ort built from sources it generates correctly the `scan-result.yml` but with docker it crashes and reports `Provenance Errors`, most of them are similar to;
```
Could not resolve provenance for package '...:': IOException: Could not resolve provenance for '...:' for source code origins [VCS, ARTIFACT].
Resolution of VCS failed with:
IOException: Could not resolve revision for package '....:' with VcsInfo(type=Git, url=https://gitlab.com/privateProject/privateRepo.git, revision=...., path=....):
	Could not find any revision candidates for package '....:' with VcsInfo(type=Git, url=https://gitlab.com/privateProject/privateRepo.git, revision=....., path=...)
```

Any clue why the docker version fails?

CC: @dgutson

**Comment by sschuberth** [comment]
> Any clue why the docker version fails?

I guess this is because inside Docker the credentials for your private repo are unknown, whereas on your host machine your probably have set up Git so that it can clone from that private repo.

**Comment by qequ** [comment]
@franco0700

**Comment by qequ** [comment]
@sschuberth do you mean setting up a .netrc file like in the [instructions](https://github.com/oss-review-toolkit/ort/blob/main/docs/hints-for-use-with-docker.md) or setting up an [.ssh key](https://stackoverflow.com/questions/23391839/clone-private-git-repo-with-dockerfile)?

**Reply by fviernau** [reply]
Also note that you are not mounting any `~/.ort/config/config.yml` into docker run, so configuration may differ compared to your non-docker run.

**Reply by jonathan1248** [reply]
The instructions link is deprecated, I guess it was moved to [here](https://oss-review-toolkit.org/ort/docs/guides/docker#authenticating-with-a-private-git-repository-fails).

**Comment by Franco0700** [comment, marked as answer]
@qequ is just setting the .netrc file correctly. Already fixed
